File: Christine/HappyNumber.java
class Solution {
    public int squareAndSum(int num) {
        int sum = 0;

        while (num > 0) {
            int digit = num % 10;
            sum += (digit * digit);
            num /= 10;
        }

        return sum;
    }
    public boolean isHappy(int n) {
        HashSet<Integer> sums = new HashSet<>();

        while (n != 1) {
            n = squareAndSum(n);

            if (sums.contains(n)) {
                return false;
            }

            sums.add(n);
        }

        return true;
    }
}

File: RIthvik/HappyNumber.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        seenNums = set()
        while n != 1:
            seenNums.add(n)
            newNum = 0
            for digit in str(n):
                digitInt = int(digit)
                newNum += (digitInt * digitInt)
            n = newNum
            if n in seenNums:
                return False
        return True
